src/main/java/pl/project13/maven/git/GitCommitIdMojo.java

```diff
@@ -525,6 +526,41 @@ void loadGitData(@NotNull Properties properties) throws GitCommitIdExecutionExce
     }
   }
 
+  /**
+   * Fill the properties file build number environment variables which are supplied by build servers
+   *  build.number is the project specific build number, if this number is not available the unique number will be used
+   *  build.number.unique is a server wide unique build number
+   * @param properties a properties instance to put the entries on
+   */
+  void loadBuildNumber(@NotNull Properties properties) {
+    String buildNumber = null;
+    String uniqueBuildNumber = null;
+
+    //JENKINS
+    if (System.getenv().containsKey("BUILD_NUMBER"))  {
+      buildNumber = System.getenv().get("BUILD_NUMBER");
```

Comment:
does this work? If yes, then please include this info in the readme -- it did not mention that jenkins also works

Reply:
IMHO the GitCommitIdMojo is already messy as it is, and doesn't need extra logic. Why not move this into the GitDataProvider? 
It has quite similar code for retrieving all sorts of environment variables if we run on a CI server.
Should we always try to extract the properties or only if `runningOnBuildServer` is true?

Reply:
In that case I suggest that all code related to `git.build..*` variables will be moved to  a new class (e.g. `BuildDataMojo.java`).         
From there we can then detect the kind of the build enviroment and not only if the plugin is running on a CI server.       
In my opinion this would clean up the code and would allow further, if wanted, extensions to the build variables.

@TheSnoozer  My idea was that somebody might want to run a script to set the buid number manually.  And the Jenkins enviroment variable ("BUILD_NUMBER") is pretty straight forward for this.